staging/vhost-device-can/README.md

```diff
@@ -0,0 +1,114 @@
+# vhost-device-can - CAN emulation backend daemon
+
+## Description
+This program is a vhost-user backend that emulates a VirtIO CAN device.
+The device's binary takes two (2) parameters:  a socket, a 'can-devices' list.
+The socket is commonly used across all vhost-devices to communicate with
+the vhost-user frontend device. The 'can-devices' represents a list of
+CAN/FD devices appears in the host system. This list includes *can_input* and
+*can_output* devices which vhost-device-can will forward messages to and from
+the frontend side.
+
+This program is tested with QEMU's `vhost-user-device-pci` device.
+Examples' section below.
+
+## Synopsis
+```
+**vhost-device-can** [*OPTIONS*]
+````
+
+## Options
+
+.. program:: vhost-device-can
+
+.. option:: -h, --help
+
+  Print help.
+
+.. option:: -s, --socket-path=PATH
+
+  Location of vhost-user Unix domain sockets, this path will be suffixed with
+  0,1,2..socket_count-1.
+
+.. option:: -c, --socket-count=INT
+
+  Number of guests (sockets) to attach to, default set to 1.
+
+.. option:: -d, --can-devices='CAN/FD interfaces'
```

Comment:
Hello, I am not familiar with CAN, what are these CAN/FD interfaces? It is related with SocketCAN? Thanks.

Reply:
Oh, I think these are the actual interfaces in the host.

Reply:
I was not able to test the reception. I ran `candump can0` in the guest and then `cansend vcan0 01a#11223344AABBCCDD` in the host. I get some traces from the daemon but nothing in the guest. What I am doing wrong? Thanks.

Reply:
You are right, sorry for that. I used to test it with CANFD messages by running `cangen -f vcan0` and I have left the CAN case unimplemented. I just fixed this bug, now what you described should run without issues (I have just tested it on my side).

Reply:
Cool, I just tried and it works, thanks.